source/fuzz/fuzzer_pass_add_parameters.cpp

```diff
@@ -91,5 +151,91 @@ uint32_t FuzzerPassAddParameters::GetNumberOfParameters(
+         "The pointer_type_id must refer to a pointer type");
+  auto function = fuzzerutil::FindFunction(GetIRContext(), function_id);
+  assert(function && "The function must be defined.");
+
+  // All of the local variable declarations are located in the first block.
+  auto block = function->begin();
+  for (auto& instruction : *block) {
+    if (instruction.opcode() != SpvOpVariable) {
+      continue;
+    }
+    if (!instruction.type_id() || instruction.type_id() != pointer_type_id) {
+      continue;
+    }
+
+    // Check if the found variable is marked with PointeeValueIsIrrelevant.
+    if (!GetTransformationContext()->GetFactManager()->PointeeValueIsIrrelevant(
+            instruction.result_id())) {
+      continue;
+    }
+    result_id = instruction.result_id();
+    break;
+  }
+
+  // No such variable was found. Apply a transformation to get one.
+  if (!result_id) {
+    uint32_t pointee_type_id = fuzzerutil::GetPointeeTypeIdFromPointerType(
+        GetIRContext(), pointer_type_id);
+    result_id = GetFuzzerContext()->GetFreshId();
+    ApplyTransformation(TransformationAddLocalVariable(
+        result_id, pointer_type_id, function_id,
+        FindOrCreateZeroConstant(pointee_type_id, true),
+        pointee_value_is_irrelevant));
+  }
+  return result_id;
+}
+
+uint32_t FuzzerPassAddParameters::FindOrCreateGlobalVariable(
+    uint32_t pointer_type_id, bool pointee_value_is_irrelevant) {
+  uint32_t result_id = 0;
+  auto pointer_type = GetIRContext()->get_type_mgr()->GetType(pointer_type_id);
+  // No unused variables in release mode.
+  (void)pointer_type;
+  assert(pointer_type->AsPointer() &&
+         "The pointer_type_id must refer to a pointer type");
+  for (auto& instruction : GetIRContext()->module()->types_values()) {
+    if (instruction.opcode() != SpvOpVariable) {
+      continue;
+    }
+    if (!instruction.type_id() || instruction.type_id() != pointer_type_id) {
+      continue;
+    }
+
+    // Check if the found variable is marked with PointeeValueIsIrrelevant.
+    if (!GetTransformationContext()->GetFactManager()->PointeeValueIsIrrelevant(
+            instruction.result_id())) {
+      continue;
+    }
+    result_id = instruction.result_id();
+    break;
+  }
+
+  // No such variable was found. Apply a transformation to get one.
+  if (!result_id) {
+    uint32_t pointee_type_id = fuzzerutil::GetPointeeTypeIdFromPointerType(
+        GetIRContext(), pointer_type_id);
+    auto storage_class = fuzzerutil::GetStorageClassFromPointerType(
+        GetIRContext(), pointer_type_id);
+    assert((storage_class == SpvStorageClassPrivate ||
+            storage_class == SpvStorageClassWorkgroup) &&
+           "The storage class must be Private or Workgroup");
+    result_id = GetFuzzerContext()->GetFreshId();
+    ApplyTransformation(TransformationAddGlobalVariable(
+        result_id, pointer_type_id, storage_class,
+        FindOrCreateZeroConstant(pointee_type_id, true),
```

Comment:
If I recall correctly, `TransformationAddGlobalVariable` requires initializer to be `0` if the storage class is `Workgroup`.

Reply:
In this fragment of code, I've used FindOrCreateZeroConstant so it seems to me that it creates a valid Workgroup variable. Do you mean 0 as in integer constant of 0?

Reply:
No, sorry for the confusion. I meant that an `OpVariable` instruction shouldn't have an initializer. Thus, we pass `0` instead of the result of `FindOrCreateZero`.

Reply:
Fixed in 007db36